docs/_interface/sync-modes.md

```diff
@@ -0,0 +1,135 @@
+assumptions. This page outlines three sync configurations for full nodes and one for light nodes.
+
+## Full nodes
+
+There are two types of full node that use different mechanisms to sync up to the head of the chain:
+
+### Snap (default)
+
+A snap sync'd node holds the most recent 128 block states in memory, so transactions in that range are always quickly 
+accessible. However, snap-sync only starts processing from a relatively recent block (as opposed to genesis 
+for a full node). Between the initial sync block and the 128 most recent blocks, the node stores occasional 
+checkpoints that can be used to rebuild the state on-the-fly. This means transactions can be traced back as 
+far as the block that was used for the initial sync. Tracing a single transaction requires reexecuting all 
+preceding transactions in the same block **and** all preceding blocks until the previous stored snapshot. 
+Snap-sync'd nodes are therefore functionally equal to full nodes, but the initial synchronization required 
+a checkpoint block to sync from instead of independently verifying the chain all the way from genesis. 
+Snap sync then only verifies the proof-of-work and ancestor-child block progression and assumes that the 
+state transitions are correct rather than re-executing the transactions in each block to verify the state 
+changes. Snap sync is much faster than full sync. To start a node with snap sync pass `--syncmode snap` at 
+startup.
+
+Snap sync starts by downloading the headers for a chunk of blocks. Once the headers have been verified, the block
+bodies and receipts for those blocks are downloaded. Next, state sync begins. In state-sync, Geth first downloads the 
+leaves of the state trie for each block without the intermediate nodes along with a range proof. The state trie is 
+then regenerated locally. The state download is the part of the snap-sync that takes the most time to complete 
+and the progress can be monitored using the ETA values in the log messages. However, the blockchain is also 
+progressing at the same time and invalidating some of the regenerated state data. This means it is also necessary 
+to have a 'healing' phase where errors in the state are fixed. It is not possible to monitor the progress of 
+the state heal because the extent of the errors cannot be known until the current state has already been regenerated. 
+The healing has to outpace the growth of the blockchain, otherwise the node will never catch up to the current state. 
+There are some hardware factors that determine the speed of the state healing (speed of disk read/write and internet 
+connection) and also the total gas used in each block (more gas means more changes to the state that have to be 
+handled).
+
+To summarize, snap sync progresses in the following sequence:
+- download and verify headers
+- download block bodies and receipts
+- download raw state data
+- regenerate state trie
+- heal state trie to account for newly arriving data
```

Comment:
Well, step 2 is parallel to (3-4-5). I'm not sure "regenerate state trie" should be it's own step, it's more of an internal/implicit part of downloading state.

Reply:
```suggestion
To summarize, snap sync progresses in the following sequence:
- download and verify headers
- download block bodies and receipts.In parallel, download raw state data and build state trie
- heal state trie to account for newly arriving data
```